notebook/static/notebook/js/savewidget.js

```diff
@@ -65,6 +65,61 @@ define([
             that.set_autosaved(data.value);
         });
     };
+    
+    /** A modal dialog that is used for the specific purpose of
+   * getting a notebook name, validating it, and passing a reference 
+   * of it to the button click handler.
+   * @param options Expected to have dialog message, dialog title,
+   * a primary button and a click handler associated with them.
+   */
+   SaveWidget.prototype._nb_name_dialog = function(options) {
+       var that = this;
+       var dialog_template = _.template(`<p class="<%= classname %>"><%= msg %></p>
```

Comment:
Neat! I assume the `_.template()` makes it translatable?

Reply:
No, I'm sorry. Underscore templates are just plain html templates.  This was just an attempt to share some of the ui code for "save as" and "rename".